src/com/twitter/intellij/pants/service/PantsCompileOptionsExecutor.java

```diff
@@ -131,15 +134,51 @@ public PantsCompileOptions getOptions() {
   }
 
   @NotNull
-  public String loadProjectStructure(
+  public List<String> loadProjectStructure(
     @NotNull Consumer<String> statusConsumer,
     @Nullable ProcessAdapter processAdapter
   ) throws IOException, ExecutionException {
     if (PantsUtil.isExecutable(getProjectPath())) {
-      return loadProjectStructureFromScript(getProjectPath(), statusConsumer, processAdapter);
+      return Collections.singletonList(loadProjectStructureFromScript(getProjectPath(), statusConsumer, processAdapter));
     }
     else {
-      return loadProjectStructureFromTargets(statusConsumer, processAdapter);
+      return loadExportDataForProject(statusConsumer, processAdapter);
+    }
+  }
+
+  @NotNull
+  private List<String> loadExportDataForProject(
+    @NotNull Consumer<String> statusConsumer,
+    @Nullable ProcessAdapter processAdapter
+  ) throws IOException, ExecutionException {
+    String[] targets = loadMinimizedTargets(statusConsumer, processAdapter);
+    Stream<String> stream = Arrays.stream(targets).map(target -> getExport(target, processAdapter));
+    return stream.collect(Collectors.toList());
+  }
+
+  @NotNull
+  private String[] loadMinimizedTargets(
+    @NotNull Consumer<String> statusConsumer,
+    @Nullable ProcessAdapter processAdapter
+  ) throws IOException, ExecutionException {
+    statusConsumer.consume("Determining optimal import set...");
+    final File outputFile = FileUtil.createTempFile("pants_minimzie_run", ".out");
+    final GeneralCommandLine commandLine = PantsUtil.defaultCommandLine(getProjectPath());
+    commandLine.addParameter("minimize");
+    commandLine.addParameter("--minimize-output-file=" + outputFile.getPath());
+    commandLine.addParameter("::");
```

Comment:
still need to use `getAllTargetAddresses` instead of hardcoding `::`, otherwise when user selects `buildroot/a/b/c::` it will still do `buildroot::`

P.S. `getAllTargetAddresses` really means getting the target specs, one of the legacies I haven't got to yet.